contracts/staking/StakingBase.sol

```diff
@@ -68,145 +64,133 @@ contract StakingBase is Ownable, ReentrancyGuard, IStakingBase {
+            // meaning we have to check which timestamp is more recent
+            uint256 mostRecentTimestamp = staker.lastTimestampLocked > staker.unlockedTimestamp
+                ? staker.lastTimestampLocked
+                : staker.unlockedTimestamp;
+
+            rewards += staker.owedRewardsLocked + _getInterimRewards(staker, block.timestamp - mostRecentTimestamp, true);
         }
 
-        emit Claimed(msg.sender, rewards, address(rewardsToken));
+        return rewards;
     }
 
-    function _getPendingRewards(
-        Staker memory staker
-    ) internal view returns (uint256) {
-        // Return any existing rewards they are owed plus the additional amount accrued
-        // Value is prorated to a fractional period length. This means that calls will 
-        // calculate rewards for the appropriate amount in between periods, instead of
-        // just the full period
-
-        // The fractional amount of a period that has passed
-        uint256 fractionOfPeriod = ((block.timestamp - staker.lastUpdatedTimestamp) % periodLength);
-
-        // Calculate rewards owed for the number of periods
-        uint256 fixedPeriodRewards = 
-                PRECISION_MULTIPLIER * (rewardsPerPeriod *
-                staker.amountStaked *
-                ((block.timestamp - staker.lastUpdatedTimestamp) /
-                    periodLength)) / PRECISION_MULTIPLIER;
-
-        // Calculate rewards owed for the fractional amount of this period
-        uint256 partialRewards = 
-            PRECISION_MULTIPLIER * 
-                ((fractionOfPeriod * rewardsPerPeriod * staker.amountStaked) /
-                    periodLength) / PRECISION_MULTIPLIER;
-
-        // Return the full period rewards prorated up to the moment they call
-        return staker.owedRewards + fixedPeriodRewards + partialRewards;
-    }
-
-    function _getContractRewardsBalance() internal view virtual returns (uint256) {
-        return rewardsToken.balanceOf(address(this));
-    }
+    // Get rewards a user accrued between their last touch point and now
+    // In the `locked` case, we only calculate rewards if the user has surpassed their lock period
+    function _getInterimRewards(Staker storage staker, uint256 timePassed, bool locked) internal view returns (uint256) {
+        if (locked) {
+            // uint256 retvalB = staker.amountStaked * (rewardsPerPeriod * (timePassed)) / periodLength / 1000;
+            // console.log("retvalB: %s", retvalB);
 
-    function _checkRewardsAvailable(uint256 rewardAmount) internal view {
-        uint256 rewardBalance = _getContractRewardsBalance();
+            return staker.amountStakedLocked * (rewardsPerPeriod * (timePassed)) / periodLength / 1000;
+        } else {
 
-        if (rewardBalance < rewardAmount || rewardBalance == 0) {
-            revert NoRewardsLeftInContract();
+            return staker.amountStaked * (rewardsPerPeriod * (timePassed)) / periodLength / 1000;
```

Comment:
can optimize here since the formula is the same, only the amout value is different, so only that amount could be under the `if`